lib/Sema/DerivedConformanceCodingKey.cpp

```diff
@@ -405,8 +405,9 @@ static bool canSynthesizeCodingKey(TypeChecker &tc, Decl *parentDecl,
       return false;
   }
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
```

Comment:
We should probably validate on demand here rather than bailing out. A plain `tc.validateDecl(enumDecl)` would avoid all this.